src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -171,70 +162,69 @@ private void initializePlaceholderSettings() {
     }
 
     private void initializeLocalListeners() {
-        // Add listener to tab flags
-        this.isLoansTab = logic.getIsLoansTab();
-        this.isAnalyticsTab = logic.getIsAnalyticsTab();
-        this.isPersonTab = logic.getIsPersonTab();
-
-        // Add listener to loaneeInfoFlag
-        this.loaneeInfoFlag = logic.getLoaneeInfoFlag();
+        this.tabIndicator = logic.getTabIndicator();
 
-        logic.setIsAnalyticsTab(false);
-        logic.setIsLoansTab(false);
-
-        this.isPersonTab.addListener((observable, oldValue, newValue) -> {
-            toggleTabs();
-        });
-        this.isLoansTab.addListener((observable, oldValue, newValue) -> {
-            toggleTabs();
-        });
-        this.isAnalyticsTab.addListener((observable, oldValue, newValue) -> {
+        this.tabIndicator.addListener((observable, oldValue, newValue) -> {
+            System.out.println("Value changed!" + newValue);
```

Comment:
This System.out.println statement probably needs to be removed!

Reply:
Oops, will remove it now